cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -2005,78 +2007,84 @@ func (p *BicepProvider) ensureParameters(
 		}
 
 		// No saved value for this required parameter, we'll need to prompt for it.
-		parameterPrompts = append(parameterPrompts, struct {
-			key   string
-			param azure.ArmTemplateParameterDefinition
-		}{key: key, param: param})
+		promptForParams = append(promptForParams, parameterPrompts{key: key, param: &param})
 	}
 
-	if len(parameterPrompts) > 0 {
-		if p.console.SupportsPromptDialog() {
+	promptedParameters, configModified, err := p.promptParameters(ctx, promptForParams)
+	if err != nil {
+		return nil, fmt.Errorf("prompting for parameters: %w", err)
+	}
 
-			dialog := input.PromptDialog{
-				Title: "Configure required deployment parameters",
-				Description: "The following parameters are required for deployment. " +
-					"Provide values for each parameter. They will be saved for future deployments.",
-			}
+	for key, value := range promptedParameters {
+		configuredParameters[key] = value
+	}
 
-			for _, prompt := range parameterPrompts {
-				dialog.Prompts = append(dialog.Prompts, p.promptDialogItemForParameter(prompt.key, prompt.param))
-			}
+	if configModified {
+		if err := p.envManager.Save(ctx, p.env); err != nil {
+			p.console.Message(ctx, fmt.Sprintf("warning: failed to save configured values: %v", err))
+		}
+	}
+	p.ensureParamsInMemoryCache = maps.Clone(configuredParameters)
+	return configuredParameters, nil
+}
 
-			values, err := p.console.PromptDialog(ctx, dialog)
-			if err != nil {
-				return nil, fmt.Errorf("prompting for values: %w", err)
-			}
+func (p *BicepProvider) promptParameters(
```

Comment:
Needs some unit tests here.  This prompting logic has become sufficiently complex and I'm afraid we are going to cause regressions.